### PR TITLE
Update various dependencies (both libraries and plugins)

### PR DESCRIPTION
**Pull request by wendigo**
Spring cleaning in progress :)

### CODE REVIEW

File: core/docker/Dockerfile [hunk 1]
```diff
@@ -38,7 +38,7 @@ RUN \
     chown -R "trino:trino" /usr/lib/trino /data/trino
 
 ARG TRINO_VERSION
-COPY trino-cli-${TRINO_VERSION}-executable.jar /usr/bin/trino
+COPY --chown=trino:trino trino-cli-${TRINO_VERSION}-executable.jar /usr/bin/trino
```
Comment (wendigo) [inline core/docker/Dockerfile:41]: This is the only significant change. Previously the file was copied as root with broader access then needed

Comment (kokosing) [reply core/docker/Dockerfile:41]: separate preparatory commit?

**Comment by wendigo** [comment]
CI hit https://github.com/trinodb/trino/issues/11140

File: pom.xml [hunk 2]
```diff
@@ -1944,7 +1944,7 @@
                 <plugin>
                     <groupId>org.skife.maven</groupId>
                     <artifactId>really-executable-jar-maven-plugin</artifactId>
-                    <version>1.0.5</version>
+                    <version>2.1.1</version>
```
Comment (kokosing) [inline pom.xml:1947]: It looks like a major change. What changed?

File: pom.xml [hunk 3]
```diff
@@ -1105,7 +1105,7 @@
             <dependency>
                 <groupId>com.github.ben-manes.caffeine</groupId>
                 <artifactId>caffeine</artifactId>
```
Comment (kokosing) [inline pom.xml:1107]: Where this is used? Can we drop it maybe?

**Review by kokosing — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (wendigo) [reply pom.xml:1947]: The way zip is made executable. Plugin works the same

File: pom.xml (hunk 3 above)
Comment (wendigo) [reply pom.xml:1107]: It is used. We can rewrite call sites

**Comment by kokosing** [comment]
Thanks!

**Pull request merged**